proposals/NNNN-async-stream-factory.md

```diff
@@ -0,0 +1,166 @@
+      self.continuation = continuation
+    }
+  }
+
+  /// Initializes a new ``AsyncStream`` and an ``AsyncStream/Continuation``.
+  ///
+  /// - Parameters:
+  ///   - elementType: The element type of the stream.
+  ///   - limit: The buffering policy that the stream should use.
+  /// - Returns: A ``NewStream`` struct which contains the stream and its continuation.
+  public static func makeStream(
+      of elementType: Element.Type = Element.self,
+      bufferingPolicy limit: Continuation.BufferingPolicy = .unbounded
+  ) -> NewStream {
+    let storage: _Storage = .create(limit: limit)
+    let stream = AsyncStream<Element>(storage: storage)
+    let continuation = Continuation(storage: storage)
+    return .init(stream: stream, continuation: continuation)
+  }
+}
+
+extension AsyncThrowingStream {
+  /// Struct for the return type of ``AsyncThrowingStream/makeStream(elementType:limit:)``.
+  ///
+  /// This struct contains two properties:
+  /// 1. The ``continuation`` which should be retained by the producer and is used
+  /// to yield new elements to the stream and finish it.
+  /// 2. The ``stream`` which is the actual ``AsyncThrowingStream`` and
+  /// should be passed to the consumer.
+  public struct NewStream: Sendable {
+    /// The continuation of the ``AsyncThrowingStream`` used to yield and finish.
+    public let continuation: AsyncThrowingStream<Element, Failure>.Continuation
+
+    /// The stream which should be passed to the consumer.
+    public let stream: AsyncThrowingStream<Element, Failure>
+
+    public init(stream: AsyncThrowingStream<Element, Failure>, continuation: AsyncThrowingStream<Element, Failure>.Continuation) {
+      self.stream = stream
+      self.continuation = continuation
+    }
+  }
+
+  /// Initializes a new ``AsyncThrowingStream`` and an ``AsyncThrowingStream/Continuation``.
+  ///
+  /// - Parameters:
+  ///   - elementType: The element type of the stream.
+  ///   - failureType: The failure type of the stream.
+  ///   - limit: The buffering policy that the stream should use.
+  /// - Returns: A ``NewStream`` struct which contains the stream and its continuation.
+  public static func makeStream(
+      of elementType: Element.Type = Element.self,
+      throwing failureType: Failure.Type = Failure.self,
+      bufferingPolicy limit: Continuation.BufferingPolicy = .unbounded
+  ) -> NewStream {
+    let storage: _Storage = .create(limit: limit)
+    let stream = AsyncThrowingStream<Element, Failure>(storage: storage)
+    let continuation = Continuation(storage: storage)
+    return .init(stream: stream, continuation: continuation)
+  }
+}
+```
+
+## Source compatibility, Effect on ABI stability, Effect on API resilience
+
+As this is an additive change, it should not have any compatibility, stability or resilience problems. The only potential problem would be if someone has already run into this shortcoming and decided to define their own `makeStream` methods.
+
+## Alternatives considered
```

Comment:
I looked through the pitch and identified a couple of alternatives you may want to add:

* [Using an initializer on `NewStream` instead of a static method that returns `NewStream`](https://forums.swift.org/t/pitch-convenience-async-throwing-stream-makestream-methods/61030/20)
* [Separately allocating the continuation and passing it into a stream `init`](https://forums.swift.org/t/pitch-convenience-async-throwing-stream-makestream-methods/61030/27)

Discussing these ideas and preemptively rebutting them helps to keep reviewers from rehashing topics you've already settled.